src/ASM/ASMs2D.C

```diff
@@ -1519,6 +1519,7 @@ bool ASMs2D::integrate (Integrand& integrand,
   PROFILE2("ASMs2D::integrate(I)");
 
   bool use2ndDer = integrand.getIntegrandType() & Integrand::SECOND_DERIVATIVES;
```

Comment:
you removed the or here. this mean you won't get hessians with only third derivative flag enabled. i for one would expect second derivatives as well when requesting third.

Reply:
That's correct. Think it is consistent that the integrand sets both flags if both 2nd and 3rd derivatives are needed, and in the unlikely event that only 3rd (but not 2nd) that can be achieved as well by only setting the 3rd derivative flag

Reply:
Providing first derivatives is the default (enum value STANDARD), as that is always needed for any integrand (can switch of that too using NO_DERIVATIVES, which is used in the ElasticBar integrand where all matrices are set up explicitly, but that's a rare exception). Setting either the 2nd or 3rd flag, will provide those derivatives _in addition_ to the 1st. So in that context it is the logic behavior, in my opinion.

Reply:
logic != consistent. anyways..

Reply:
logic, consistent, whatever.. This convention is OK then?

Reply:
yeah sure, it's fine, just, consistency was why i did as i did was my point.